### ISSUE TITLE
Unboxing from a branch fails when a slash is in the branch name

### ISSUE THREAD

**Issue by adrianmcli** [trufflesuite/truffle#1676]
## Issue

This is the syntax for unboxing from a branch:

```
truffle unbox owner-name/repo-name#branch-name
```

However, when `branch-name` has a forward slash in it, unboxing fails with an error:

```
Error: Box specified in invalid format
```

This is where the problem is: https://github.com/trufflesuite/truffle/blob/e913be03c564cd967d58a67ddc16360188db3ead/packages/truffle-core/lib/commands/unbox.js#L7-L34

## Steps to Reproduce

Try to unbox with the following command:

```
truffle unbox truffle-box/webpack-box#fix/fail-to-unbox
```

## Expected Behavior

It should just unbox normally.

## Actual Results

Error stack:

```
Error: Box specified in invalid format
    at normalizeURL (/Users/adrianli/n/lib/node_modules/truffle/build/webpack:/packages/truffle-core/lib/commands/unbox.js:33:1)
    at Object.run (/Users/adrianli/n/lib/node_modules/truffle/build/webpack:/packages/truffle-core/lib/commands/unbox.js:123:1)
    at Command.run (/Users/adrianli/n/lib/node_modules/truffle/build/webpack:/packages/truffle-core/lib/command.js:113:1)
    at Object.<anonymous> (/Users/adrianli/n/lib/node_modules/truffle/build/webpack:/packages/truffle-core/cli.js:47:1)
    at __webpack_require__ (/Users/adrianli/n/lib/node_modules/truffle/build/webpack:/webpack/bootstrap 10c8e5c48cca88d8ca36:19:1)
    at /Users/adrianli/n/lib/node_modules/truffle/build/webpack:/webpack/bootstrap 10c8e5c48cca88d8ca36:65:1
    at Object.<anonymous> (/Users/adrianli/n/lib/node_modules/truffle/build/cli.bundled.js:71:10)
    at Module._compile (internal/modules/cjs/loader.js:688:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:699:10)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:741:12)
    at startup (internal/bootstrap/node.js:285:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:739:3)
Truffle v5.0.0 (core: 5.0.0)
Node v10.13.0
```

## Environment

Truffle v5.0.0 (core: 5.0.0)
Node v10.13.0

**Comment by eggplantzzz** [comment]
Hmm, there also seems to be an issue with the github-download package. I filed an issue (here)[https://github.com/jprichardson/node-github-download/issues/15]. It seems like it is currently a limitation but we'll see if he plans to fix it.

**Comment by adrianmcli** [comment]
@eggplantzzz why don't we use this one instead: https://github.com/flipxfx/download-git-repo

I recently used it with my [Nano React App](https://github.com/adrianmcli/nano-react-app) project, and it:

- has way more stars (372 vs 53)
- is updated in the last year (instead of in 2016)

**Comment by eggplantzzz** [comment]
Let me take a look and see how easy it will be to swap them out, I'm not opposed to switching.

**Comment by eggplantzzz** [comment]
Fixed in version 5.3.11.
